sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -518,8 +520,32 @@ case class DescribeTableCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val result = new ArrayBuffer[Row]
-    val catalog = sparkSession.sessionState.catalog
+    if (query.isDefined) {
+      query.get match {
+        case _: InsertIntoTable | _: InsertIntoDir =>
+          throw new AnalysisException("Describe command is not supported for insert statements.")
+        case u @ Union(children) if children.forall(_.isInstanceOf[InsertIntoTable]) =>
+          throw new AnalysisException("Describe command is not supported for insert statements.")
+        case u: UnresolvedRelation =>
```

Comment:
Can we put the two matches above together?
```
        case p if p.collectFirst { case _: InsertIntoTable | _: InsertIntoDir => true }.isDefined =>
```

Reply:
@maropu This is fantastic :-). Let me try it.